dashboard/modules/log/log_agent.py

```diff
@@ -279,7 +373,20 @@ async def StreamLog(self, request, context):
                 start_offset = 0
                 end_offset = find_end_offset_file(f)
 
-                if lines != -1:
+                if task_id is not None:  # Stream from task log.
+                    attempt_number = (
+                        request.attempt_number
+                        if request.HasField("attempt_number")
+                        else 0
+                    )
+                    start_offset, end_offset = await self._find_task_log_offsets(
```

Comment:
Hmm maybe depending on the overhread of file APIs, we can consider offloading them to a different thread not to bother main dashboard ops.

Reply:
yeah, let me do the benchmarking.

Reply:
Streaming 3GiB data will take 3 seconds. Current approach is offloading the routine into another thread to unblock the event loop.

Reply:
This sgtm